### ISSUE TITLE
[Usability] User can't identify an element or its state.

### ISSUE THREAD

**Issue by shiragoodman** [department-of-veterans-affairs/va.gov-team#80111]
_This ticket was generated during the pilot process for the new VA.gov experience standards. Please tag `@platform-governance-team-members` on Slack if you need assistance interpreting this ticket._

## Product Information
**Team:** Benefits Non-Disability
**Product:** Burial Form 530
**Feature:** 21P-530 Parity Updates

## Findings details
**VA.gov Experience Standard - issue:** User can't identify an element or its state.
**VA.gov Experience Standard - category:** Usability
**Launch-blocking:** No
**Design System review:** No
**Collab Cycle Reviewer:** @briandeconinck (Accessibility)

### Description

For the list and loops on [v2 /military-history/service-periods](https://staging.va.gov/burials-and-memorials-v2/application/530/military-history/service-periods) and [v2 /military-history/previous-names/add](https://staging.va.gov/burials-and-memorials-v2/application/530/military-history/previous-names/add), when you have multiple entries, the Edit buttons need to have unique accessible names.

Screen reader users will often tab through the interactive elements of a page, or otherwise have their screen reader announce a list of all of the buttons or links on the page. When multiple elements have identical names (eg. "Edit previous name" or "Edit service period"), it's hard to distinguish them from each other --- and even when exploring the surrounding context, it takes some effort to parse out which item each edit button is associated with.

<details><summary>Link, screenshot or steps to recreate</summary>

As currently coded, the Edit buttons look something like:
```
<button type="button" class="usa-button-secondary edit vads-u-flex--auto" aria-label="Edit previous name">Edit</button>
```

</details>

## Recommended action

I think the best option would be something like `aria-label="Edit [name]"`, eg. `aria-label="Edit Brian DeConinck"`. If that's not possible, then some index value like `aria-label="Edit previous name 1"` would work too.

Likewise for service periods --- `Edit service period 1`, for example, since some Veterans may have multiple service periods in the same branch.

## References

- **WCAG Success Criteria:** [WCAG 2.4.9 Link Purpose (Link Only) - Level AAA](https://www.w3.org/TR/WCAG21/#link-purpose-link-only)
- **Modality:** Screen reader
- **Design System Component:** N/A
- **Design System Pattern or Template:** N/A
- **Design System Foundation:** N/A
- **Content Style Guide:** N/A
- **Context:** N/A
- **Accessibility Defect Severity:** [2: High. Should be fixed in 1-2 sprints post-launch.](https://depo-platform-documentation.scrollhelp.site/developer-docs/accessibility-defect-severity-rubric#Accessibilitydefectseverityrubric-a11y-defect-2)

---

### Next Steps for the VFS Team

- [ ] **Questions?** For the most timely response, comment on Slack in your team channel tagging `@platform-governance-team-members` with any questions or to get help validating the issue.
- [ ] **Ticket ownership:** If you believe that this issue is out of scope, not your team's responsibility, or a Design System issue, comment and tag `@platform-governance-team-members` on your team channel in Slack to provide an explanation and who you believe is responsible. The Governance team will follow up.
- [ ] **Close this ticket** when the issue has been resolved or validated by your Product Owner.

**Comment by DLarson-Oddball** [comment]
@steele-lm Code for this issue has been merged.
